kernel/src/process.rs

```diff
@@ -1528,6 +1570,15 @@ impl<C: Chip> ProcessType for Process<'a, C> {
     }
 
     unsafe fn print_full_process(&self, writer: &mut dyn Write) {
+        if config::CONFIG.debug_processes == false {
+            let _ = writer.write_str(
+                "\
+                \r\nWARNING: Values reported below are inaccurate because\
+                \r\nprocess debugging is currently disabled. This setting\
+                \r\ncan be enabled in `kernel/src/config.rs`.\
+                \r\n",
+            );
+        }
```

Comment:
If `debug_processes` is disabled, could this function simply write this banner and skip all of the rest? This would also reduce code size because the rest of the function would be dead code.

Also, will this banner be printed once per process, or once for all processes?

Reply:
> If debug_processes is disabled, could this function simply write this banner and skip all of the rest? This would also reduce code size because the rest of the function would be dead code.

It could, but this function also prints some other useful information that is not affected by `debug_processes`, e.g. stored state of the process and MPU config.


> Also, will this banner be printed once per process, or once for all processes?

This banner will be printed per process. I can move the banner to `panic_process_info` since it is the only caller for now but I thought keeping it here would be safer, just in case this function is called from somewhere else in the future. It is also harder to miss when there is a lot of debug output.

Please let me know what you think.

Reply:
I thought that printing the banner once would be enough, and avoid verbosity, but if you think it's better to print it multiple times to avoid missing it I don't have a strong opinion either.